verilog/analysis/verilog_linter.cc

```diff
@@ -206,28 +210,53 @@ std::vector<LintRuleStatus> VerilogLinter::ReportStatus(
   return statuses;
 }
 
-LinterConfiguration LinterConfigurationFromFlags() {
-  LinterConfiguration config;
-
-  // Turn on default ruleset.
-  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
-  config.UseRuleSet(ruleset);
+const std::string linter_config_file = ".rules.verible_lint";
 
+bool AppendLinterConfigurationFromFile(
+    LinterConfiguration &config, absl::string_view config_filename) {
   // Read local configuration file
   std::string content;
-  if (verible::file::GetContents(absl::GetFlag(FLAGS_rules_config), &content)) {
+  if (verible::file::GetContents(config_filename, &content)) {
     RuleBundle local_rules_bundle;
     std::string error;
     if (local_rules_bundle.ParseConfiguration(content, &error)) {
       config.UseRuleBundle(local_rules_bundle);
     } else {
-      LOG(ERROR) << "Unable to fully parse configuration: " << error
-                 << std::endl;
+      LOG(WARNING) << "Unable to fully parse configuration: " << error;
+    }
+    return true;
+  }
+
+  return false;
+}
+
+LinterConfiguration LinterConfigurationFromFlags(absl::string_view filename) {
+  LinterConfiguration config;
+
+  // Turn on default ruleset.
+  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
+  config.UseRuleSet(ruleset);
+
+  if (FLAGS_rules_config.IsModified()) {
+    // Use configuration file from flag if specified
+    std::string config_file = absl::GetFlag(FLAGS_rules_config);
+
+    if (!AppendLinterConfigurationFromFile(config, config_file)) {
+      LOG(WARNING) << "Unable to open linter configuration file: "
+                   << config_file;
+    }
+  } else if (absl::GetFlag(FLAGS_rules_config_search)) {
+    // Search upward if search is enabled and no configuration file is
+    // specified
+    std::string config_file;
```

Comment:
`resolved_config_file` ?  (name these to make clear the relationship between filenames)